components/TransactionCard/index.js

```diff
@@ -1,23 +1,34 @@
 import React, { useState } from "react";
 import * as Styled from "./TransactionCard.styled";
 import ModalDelete from "../ModalDelete";
+import Modal from "../Modal";
 import { categories } from "@/utils/categories";
+import TransactionEntryForm from "../TransactionEntryForm";
 
 function TransactionCard({ transaction, deleteTransaction }) {
-  const [showModal, setShowModal] = useState(false);
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+  const [showEditModal, setShowEditModal] = useState(false);
+
   const currentCategory = categories.find(
     (category) => category.name === transaction.category,
   );
 
   return (
+    <>
     <Styled.CardContainer
       key={transaction._id}
       category={transaction.category}
       $color={currentCategory.softColor}
-    >
+    >   
+    <Modal showModal={showEditModal}>
+    <TransactionEntryForm
+setShowModal={setShowEditModal}
+currentTransaction={transaction}
+/></Modal>
+
       <ModalDelete
-        showModal={showModal}
-        setShowModal={setShowModal}
+        showModal={showDeleteModal}
+        setShowModal={setShowDeleteModal}
         deleteTransaction={deleteTransaction}
         id={transaction._id}
       />
```

Comment:
Why do you have a Modal and a ModalDelete Component? Maybe you can reuse the Modal component here.